packages/main/src/plugin/extension/extension-loader.ts

```diff
@@ -1738,18 +1732,17 @@ export class ExtensionLoader {
       return;
     }
 
-    const telemetryOptions = { extensionId: extension.id };
+    const telemetryOptions: { extensionId: string; error?: unknown } = { extensionId: extension.id };
 
     this.extensionState.set(extension.id, 'stopping');
     this.apiSender.send('extension-stopping');
 
     if (extension.deactivateFunction) {
       try {
         await extension.deactivateFunction();
```

Comment:
@jeffmaury would this approach work? This way you do not need to check if function is async.

```
await Promise.resolve(extension.deactivateFunction());
```